### PR TITLE
New Keyboard: Quantrik Kyuu 65% Board

### PR DESCRIPTION
**Pull request by mechmerlin**
## Description

Port over the Quantrik Kyuu 65% into QMK from a known good kbfirmware.json file courtesy of Chimera. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/quantrik/kyuu/config.h [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+#define MATRIX_COL_PINS { F1, F4, F5, F6, F7, C7, C6, F0, B7, D0, D5, D3, D2, D1, B3 }
+#define UNUSED_PINS
+
+/* COL2ROW, ROW2COL*/
+#define DIODE_DIRECTION COL2ROW
+
+/*
+ * Split Keyboard specific options, make sure you have 'SPLIT_KEYBOARD = yes' in your rules.mk, and define SOFT_SERIAL_PIN.
+ */
+#define SOFT_SERIAL_PIN D0 // or D1, D2, D3, E6
```
Comment (drashna) [inline keyboards/quantrik/kyuu/config.h]: Is this a split? 

If not: 

```suggestion
// #define SOFT_SERIAL_PIN D0 // or D1, D2, D3, E6
```

File: keyboards/quantrik/kyuu/config.h [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xFEED
+#define PRODUCT_ID      0x0000
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    Quantrik
+#define PRODUCT         kyuu
```
Comment (vomindoraan) [inline keyboards/quantrik/kyuu/config.h]:
```suggestion
#define PRODUCT         Kyuu
```

File: keyboards/quantrik/kyuu/config.h (hunk 1 above)
Comment (mechmerlin) [reply keyboards/quantrik/kyuu/config.h]: Wait what? Why is this default?

Comment (drashna) [reply keyboards/quantrik/kyuu/config.h]: Not sure.  Though, it's harmless, if you don't enable the split keyboard settings.

File: keyboards/quantrik/kyuu/keymaps/default/readme.md [hunk 3]
```diff
@@ -0,0 +1 @@
+# The default keymap for kyuu
```
Comment (vomindoraan) [inline keyboards/quantrik/kyuu/keymaps/default/readme.md]:
```suggestion
# The default keymap for Kyuu
```

File: keyboards/quantrik/kyuu/info.json [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+          {"label":"K46 (D6,C6)", "x":4, "y":4, "w":7},
+          {"label":"K4A (D6,D5)", "x":11, "y":4, "w":1.5},
+          {"label":"K4B (D6,D3)", "x":13, "y":4},
+          {"label":"K4C (D6,D2)", "x":14, "y":4},
+          {"label":"K4D (D6,D1)", "x":15, "y":4}
+        ]
+      }
+    }
+  }
+
```
Comment (vomindoraan) [inline keyboards/quantrik/kyuu/info.json]: The indentation here looks a bit wonky. The first level is 4 spaces while the rest are 2. You could make it consistent by deleting the first two columns on lines 2–81 (i.e. reducing the indentation by 2 spaces).

Comment (mechmerlin) [reply keyboards/quantrik/kyuu/info.json]: you'll have to take that up with noroadsleft.

File: keyboards/quantrik/kyuu/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+// Defines the keycodes used by our macros in process_record_user
+enum custom_keycodes {
+  QMKBEST = SAFE_RANGE,
+  QMKURL
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT( \
+    KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSLS, KC_GRV,  \
```
Comment (zvecr) [inline keyboards/quantrik/kyuu/keymaps/default/keymap.c]: Layout looks offset like its matching electrical, and wrapping too early

File: keyboards/quantrik/kyuu/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    KC_HOME, KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSPC, \
+    KC_PGUP, KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,  KC_PGDN, \
+    KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_LSFT, KC_UP,   KC_END,  KC_LCTL, \
+    KC_LGUI, KC_LALT, KC_SPC,  KC_RGUI, KC_LEFT, KC_DOWN, KC_RGHT  \
+  ),
+  [1] = LAYOUT( \
```
Comment (zvecr) [inline keyboards/quantrik/kyuu/keymaps/default/keymap.c]: Currently no way to get to layer 1, to be able to use one of the 2 `RESET` keys. Layer 1 also has the "wrapping too early" issue.

File: keyboards/quantrik/kyuu/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, \
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, RESET,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, RESET,   \
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, \
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, \
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS  \
+  ),
+
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
```
Comment (zvecr) [inline keyboards/quantrik/kyuu/keymaps/default/keymap.c]: While part of the default template, the macros are not referenced, I guess they should be removed or added to a layer?

File: keyboards/quantrik/kyuu/keymaps/default/keymap.c (hunk 5 above)
Comment (mechmerlin) [reply keyboards/quantrik/kyuu/keymaps/default/keymap.c]: Thanks!

File: keyboards/quantrik/kyuu/keymaps/default/keymap.c (hunk 6 above)
Comment (mechmerlin) [reply keyboards/quantrik/kyuu/keymaps/default/keymap.c]: Thanks! I guess the kbfirmware.json default neglected this.

File: keyboards/quantrik/kyuu/keymaps/default/keymap.c (hunk 7 above)
Comment (mechmerlin) [reply keyboards/quantrik/kyuu/keymaps/default/keymap.c]: Preserving this as an example for people who want to learn to use macros.

File: keyboards/quantrik/kyuu/info.json (hunk 4 above)
Comment (noroadsleft) [reply keyboards/quantrik/kyuu/info.json]: My parser uses 2-space indent for `info.json`. It probably got modified by the IDE, .editorconfig or something similar.

**Comment by drashna** [comment]
Thanks!

**Pull request merged**